src/rational.cpp

```diff
@@ -1,230 +1,194 @@
-#include "rational.h"
+#include "Rational.h"
 
-int getGcd(int a, int b) {
-  while (b != 0)  {
-    int t = b;
-    b = a % b;
-    a = t;
-  }
-  return a;
-}
-
-Rational::Rational(int numerator, int denominator)
-{
-    numer = numerator;
-    denom = denominator;
+int getGreatestCommonDivisor(int numerator, int denominator) {
+    while (denominator != 0) {
+        int t = denominator;
+        denominator = numerator % denominator;
+        numerator = t;
+    }
+    return numerator;
+}
+
+Rational::Rational(int numerator, int denominator) {
+    if (denominator == 0) {
+        throw "The object could not be created because of the denominator of zero!";
+    } else {
+        if (numerator < 0 && denominator < 0) {
```

Comment:
`if (denominator <0)` instead of `if (numerator < 0 && denominator < 0)` will be suficient